compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -1289,6 +1428,136 @@ void OMR::ValuePropagation::transformArrayCopyCall(TR::Node *node)
+               n4n         iconst 0
+               n52n      astore  <temp slot 5>[#431  Auto]
+               n5n         aload  <parm 1 [LSomeInterface;>[#420  Parm]
+               n54n      istore  <temp slot 6>[#432  Auto]
+               n6n         iconst 0
+    prevTT---> n56n      istore  <temp slot 7>[#433  Auto]
+               n7n         iload  TestSystemArraycopy4.ARRAY_SIZE I[#421  Static]
+  _curTree---> n9n       treetop
+               n8n         call  java/lang/System.arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
+               n3n           ==>aload
+               n4n           ==>iconst 0
+               n5n           ==>aload
+               n6n           ==>iconst 0
+               n7n           ==>iload
+    nextTT---> ...
+               ...
+               ...
+ slowBlock---> n39n      BBStart <block_-1>
+               n41n      treetop
+               n42n        call  java/lang/System.arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V [#428  final native static Method] [flags 0x20500 0x0 ] (dontTransformArrayCopyCall )
+               n49n          aload  <temp slot 3>[#429  Auto]
+               n51n          iload  <temp slot 4>[#430  Auto]
+               n53n          aload  <temp slot 5>[#431  Auto]
+               n55n          iload  <temp slot 6>[#432  Auto]
+               n57n          iload  <temp slot 7>[#433  Auto]
+               n40n      BBEnd </block_-1>
+            */
+
+         // Create the block that contains the System.arraycopy call which will be the slow path
+         TR::Block *slowBlock = TR::Block::createEmptyBlock(node, comp(), UNKNOWN_COLD_BLOCK_COUNT);
+         slowBlock->setIsCold(true);
+
+         TR::Node *newCallNode = node->duplicateTree(false); // No need to duplicate the children of the call since they will be replaced next with temp load
+         // It is important to set setDontTransformArrayCopyCall on the new call node. Otherwise, the new call node could get
+         // transformed again by VP
+         newCallNode->setDontTransformArrayCopyCall();
+
+         TR::Node *ttNode = TR::Node::create(TR::treetop, 1, newCallNode);
+         TR::TreeTop *newCallTree = TR::TreeTop::create(comp());
+         newCallTree->setNode(ttNode);
+
+         slowBlock->append(newCallTree);
+
+         TR::Node *oldCallNode = node;
+         if (trace())
+            traceMsg(comp(),"Creating temps for children of the original call node n%dn %p. new call node n%dn %p\n", oldCallNode->getGlobalIndex(), oldCallNode, newCallNode->getGlobalIndex(), newCallNode);
+
+         // Create temporaries for System.arraycopy arguments and replace the children of the new call node with the temps
+         for (int32_t i = 0 ; i < oldCallNode->getNumChildren(); ++i)
+            {
+            TR::Node *child = oldCallNode->getChild(i);
+            TR::SymbolReference *newSymbolReference = comp()->getSymRefTab()->createTemporary(comp()->getMethodSymbol(), child->getDataType());
```

Comment:
I think the temporary is only used if `isLoadConst()` is false, so I think this ought to be moved under the `else` below.

Reply:
Adressed in 71ddb70